Scripts/RemediationScripts/Remediate-EnableEncryptionInTransitForStorageAccounts.ps1

```diff
@@ -323,147 +413,269 @@ function Enable-StorageEncryptionInTransit
+                    $logResource.Add("ResourceName",($_.StorageAccountName))
+                    $logRemediatedResources += $logResource
+                    Write-Host "Successfully enabled secure transfer on the Storage Account." -ForegroundColor $([Constants]::MessageType.Update)
+                    Write-Host $([Constants]::SingleDashLine)
+                }   
+                else
+                {   
+                    $remediationFailure += $_
+                    $logResource = @{}
+                    $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+                    $logResource.Add("ResourceName",($_.StorageAccountName))
+                    $logResource.Add("Reason","Unsuccessful in enabling secure transfer on the Storage Account.")    
+                    $logSkippedResources += $logResource
+                    Write-Host "Unsuccessful in enabling secure transfer on the Storage Account." -ForegroundColor $([Constants]::MessageType.Warning) 
+                    Write-Host $([Constants]::SingleDashLine)
                 }
+            }        
+            catch
+            {
+                Write-Host "Error occurred while enabling secure transfer on the Storage Account." -ForegroundColor $([Constants]::MessageType.Error) 
+                Write-Host "Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+                Write-Host $([Constants]::SingleDashLine)
+                $remediationFailure += $_
+                $logResource = @{}
+                $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+                $logResource.Add("ResourceName",($_.StorageAccountName))
+                $logResource.Add("Reason","Error occurred while enabling secure transfer on the Storage Account.")    
+                $logSkippedResources += $logResource
             }
-            else
+        }
+        Write-Host $([Constants]::DoubleDashLine)
+        if($AutoRemediation)
+        {
+            if($($remediationSuccess|Measure-Object).Count -gt 0)
             {
-                Write-Host " $([Constants]::SingleDashLine)"
-
-                # Creating the log file
-                Write-Host "Exporting configurations of Storage Account(s) that don't have secure transfer enabled. You may want to use this CSV as a pre-check before actual remediation." 
-                $stgWithDisableHTTPS | Export-CSV -Path "$($folderpath)\StorageWithDisableHTTPS.csv" -NoTypeInformation
-                Write-Host "Path: $($folderPath)\StorageWithDisableHTTPS.csv" -ForegroundColor $([Constants]::MessageType.Info)
-            }          
-        }     
-        else
+                # Write this to a file.
+                $storageAccountRemediatedFile = "$($backupFolderPath)\RemediateStorageAccountsEnableSecureTransfer.csv"
+                $remediationSuccess | Export-CSV -Path $storageAccountRemediatedFile -NoTypeInformation
```

Comment:
One line summary like this is the list of remediated resource will help to set context for the end user.

Reply:
That is given in the Remediation summary part for standalone execution. With feature, it is not needed.

Reply:
Ok, agreed. But one thing here we are saying 'This information', how user will get what 'This information' means here.

Reply:
Corrected